internal/cli/streams/processor/delete.go

```diff
@@ -0,0 +1,74 @@
+package processor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type DeleteOpts struct {
+	cli.ProjectOpts
+	*cli.DeleteOpts
+	streamsInstance string
+	store           store.ProcessorDeleter
+}
+
+func (opts *DeleteOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *DeleteOpts) Run() error {
+	return opts.Delete(opts.store.DeleteStreamProcessor, opts.ProjectID, opts.streamsInstance)
+}
+
+// atlas streams processor delete <processorName>.
+func DeleteBuilder() *cobra.Command {
+	opts := &DeleteOpts{
+		DeleteOpts: cli.NewDeleteOpts("Atlas Stream Processor '%s' deleted\n", "Atlas Stream Processor not deleted"),
+	}
+	cmd := &cobra.Command{
+		Use:   "delete <processorName>",
+		Short: "Delete a specific Atlas Stream Processor in a Stream Processing Instance.",
+		Long:  fmt.Sprintf(usage.RequiredRole, "Project Read Only"),
```

Comment:
This role doesn't seem correct.... I don't think you should be able to delete with a read only role.